src/test/fuzz/process_message.cpp

```diff
@@ -100,7 +100,6 @@ void fuzz_target(FuzzBufferType buffer, const std::string& LIMIT_TO_MESSAGE_TYPE
         g_setup->m_node.peerman->SendMessages(&p2p_node);
     }
     SyncWithValidationInterfaceQueue();
-    LOCK2(::cs_main, g_cs_orphans); // See init.cpp for rationale for implicit locking order requirement
     g_setup->m_node.connman->StopNodes();
```

Comment:
I think there probably should be a comment as to why explicitly calling `StopNodes` is still necessary

(I believe it's because if you don't do that, then `m_node` will get destructed, deleting its `peerman`, then attempting to delete it's `connman` which will see some entries in `vNodes` and try to call `peerman->FinalizeNode()` on them, but `peerman` is deleted at that point. With the lock order fixed, it may be possible to have `~NodeContext` call `if (peerman) { peerman->Stop(); }` an remove those lines from the tests entirely. Not sure if that would also let you remove it from init.cpp)

Reply:
I think changing the destruction order can be done in a separate pull?